### PR TITLE
Undo Group Removal + Misc UX Fixes

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:* /bcgov/entity#13964

*Description of changes:*
* Implements functionality to undo group removals
* fix edit of new owner groups ownership allocation bug
* fix hide/show fee summary bug - /bcgov/entity#14298
* misc template clean up

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Review by owillborn — APPROVED** [review]
LVGTM

File: ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnersTable.vue [hunk 1]
```diff
@@ -14,7 +14,7 @@
     >
       <template
         v-slot:header
-        v-if="isMhrTransfer && hasRemovedAllHomeOwners(homeOwners) && addedOwnerCount === 0 && !hideRemovedOwners"
+        v-if="isMhrTransfer && !hasActualOwners(homeOwners) && homeOwners.length > 0"
```
Comment (ChasBoucher) [inline ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnersTable.vue:17]: Nice simplification

**Review by dimak1 — APPROVED** [review]
Nice work. Approved.

**Review by ChasBoucher — APPROVED** [review]
I was not sure of all specs, so I pulled pr and did some testing... 
Looks good.

**Pull request merged**